feat: Update pasword modal

DanielEJaimes opened the pull request:
Update password modal

DanielEJaimes committed:
feat: update password

Update password modal

DanielEJaimes committed:
fix: formData is defined but never used

Fixed lint problem

PedroChaparro reviewed (changes requested):
This feature is ready in the [python-proxy](https://github.com/hawks-atlanta/proxy-python), so you can implement the feature (not just the UI) in this PR.

DanielEJaimes committed:
feat: python-proxy implementation

The proxy response with "token is missing" but i don't know what token is required for this

DanielEJaimes commented:
The missing token is the session token?
The req body show oldpassword and newpassword
![image](https://github.com/hawks-atlanta/frontend-react/assets/131629166/31bfbcc7-06cc-4fc4-b82a-d2c4205b17b3)
The res shows only a msg 
![image](https://github.com/hawks-atlanta/frontend-react/assets/131629166/b4f63d1c-dc1d-4a8a-8641-a7e77c6d69e5)

DanielEJaimes committed:
fix: lint prefer-const

PedroChaparro commented:
> The missing token is the session token? 
> 

Yes, the token is supposed to be sent as a `Bearer` token in the `Authorization` header. Check out [this file](https://github.com/hawks-atlanta/frontend-react/blob/dev/src/services/folder/new-folder.service.ts).

DanielEJaimes committed:
feat: python-proxy implementation

PedroChaparro commented on src/components/Nav/Nav.tsx:
```diff
@@ -75,6 +84,12 @@ export function NavbarScreen() {
             )}
           </div>
+          {showProfileMenu && (
+            <div className="absolute left-0 right-0 top-full z-50 flex flex-col items-center justify-center gap-4 border bg-white p-4">
```
You copied the mobile menu styles. Those styles are meant for mobile layouts. It's not common to have menus taking the full width on desktop layouts.

Consider just adding some buttons to the navbar.

DanielEJaimes replied:
I made the menu smaller 
![image](https://github.com/hawks-atlanta/frontend-react/assets/131629166/8563c24b-15d6-4511-85f4-73c0a925a696)

PedroChaparro commented on src/components/UpdatePassword/UpdatePasswordButton.tsx:
```diff
@@ -0,0 +1,129 @@
+        Update Password
+      </button>
+      {showModal ? (
+        <div className="absolute top-full mt-4 flex w-4/5 flex-col items-center justify-center gap-4 border bg-white p-4">
```
The modal is huge. It's supposed to be a square-shaped modal according to your own designs.

![image](https://github.com/hawks-atlanta/frontend-react/assets/62714297/4624e570-13e2-401b-8311-2fd9c2bab7fe)

Make it look similar to the design, but according to the current page colors and language.

PedroChaparro replied:
Actually, there is a [dialog component](https://github.com/hawks-atlanta/frontend-react/blob/dev/src/components/Dialog.tsx) in this codebase; use it.

DanielEJaimes replied:
![image](https://github.com/hawks-atlanta/frontend-react/assets/131629166/e86f859e-ba3f-4e35-9867-91c02a0262d1)

PedroChaparro commented on src/components/UpdatePassword/UpdatePasswordButton.tsx:
```diff
@@ -0,0 +1,129 @@
+                  value: true,
+                  message: "Password required"
+                },
+                pattern: {
```
Names and labels are so confusing. Since this is a modal to update the password:

- What should users enter in the password field, is it the new password or their current password?
- What should users enter in the "Confirm password" field, is it a confirmation of the other field or a confirmation of their new password?"

DanielEJaimes replied:
You are right, it is supposed to be the current password and the new password.

PedroChaparro commented on src/components/UpdatePassword/UpdatePasswordButton.tsx:
```diff
@@ -0,0 +1,129 @@
+            <p className="m-2 ml-2 text-center text-2xl font-bold text-blue-600">
+              Update Password
+            </p>
+            <input
```
As the inputs does not have a visible `<label />`, consider using an `aria-label` attribute.

PedroChaparro reviewed (changes requested):
Test are missing too.

DanielEJaimes committed:
fix: test and style

Test added and style modified

DanielEJaimes committed:
revert: package.json faker-js dependency

DanielEJaimes committed:
fix: test

DanielEJaimes commented:
@PedroChaparro i need help here, i'm not sure what is the problem with the test
![image](https://github.com/hawks-atlanta/frontend-react/assets/131629166/5da95de0-979e-4fb4-bf0c-3b9efbf2bce7)

PedroChaparro commented:
> @PedroChaparro i need help here, i'm not sure what is the problem with the test 
> 

First of all, don't use the production URL (http://capyfile1...) for development or testing purposes. You are supposed to use the Python Proxy locally (http://localhost:8000), which is why we created the `docker-compose` file.

Probably the error is caused because of that. Also, you are not using random values for the username and password fields, so your tests only work when they are run for the first time with an empty database. That could be another cause of the error.

DanielEJaimes committed:
fix: Cypress test

DanielEJaimes committed:
fix: ERR_PNPM_OUTDATED_LOCKFILE

i used use "npm install --no-frozen-lockfile"

DanielEJaimes commented:
@PedroChaparro i'm not sure what´s the problem in here now

PedroChaparro commented on package-lock.json:
Remove this file. This repository uses `pnpm` instead of `npm`.

PedroChaparro replied:
It seems that you installed a package with `npm` so the `pnpm` lock file (`pnpm-lock.yaml`) is outdated. Install the same package with `pnpm`.

DanielEJaimes committed:
fix: pnpm dependency installed

DanielEJaimes committed:
Merge branch 'dev' into feat/updatepassword

DanielEJaimes committed:
fix: cypress test

DanielEJaimes committed:
Merge branch 'feat/updatepassword' of https://github.com/hawks-atlanta/frontend-react into feat/updatepassword

DanielEJaimes committed:
fix: pnpm package

PedroChaparro commented on src/components/Nav/Nav.tsx:
```diff
@@ -75,6 +84,15 @@ export function NavbarScreen() {
               </Fragment>
             )}
           </div>
+          {showProfileMenu && (
```
There is no way to open the modal if the user is on a phone

DanielEJaimes replied:
![image](https://github.com/hawks-atlanta/frontend-react/assets/131629166/cc419792-d644-41ca-8821-257d76f7c1c8)

PedroChaparro commented on src/components/UpdatePassword/UpdatePasswordButton.tsx:
```diff
@@ -0,0 +1,136 @@
+        Update Password
+      </button>
+      {showModal ? (
+        <div className="fixed inset-0 z-10 flex items-center justify-center backdrop-blur-sm">
```
You copied the `Dialog` component styles, use the `Dialog` component itself then.

DanielEJaimes replied:
the title is in white color but I assume it was changed in later PRs
![image](https://github.com/hawks-atlanta/frontend-react/assets/131629166/e1ac2ce8-9beb-4af7-9521-b31e1377afd0)

PedroChaparro replied:
@DanielEJaimes no, it wasn't. It is because of the `text-white` in the `Nav.tsx` file. Consider adding the `text-black` or `text-blue-600` `className` to the title in the `Dialog.tsx` file.

DanielEJaimes replied:
Made it blue-600

PedroChaparro commented on src/components/UpdatePassword/UpdatePasswordButton.tsx:
```diff
@@ -0,0 +1,136 @@
+      token: token
+    };
+    if (!errors.password && !errors.newPassword) {
+      try {
```
The `try-catch` is not needed. Since you're using a `try-catch` in the service any error will be caught there.

DanielEJaimes replied:
Done

PedroChaparro commented on src/services/auth/updatepassword.service.ts:
```diff
@@ -0,0 +1,53 @@
+      }
+    );
+
+    if (response.status !== 200) {
```
I think this `if-else` statement is not needed. If the response does not have a `200` status code an error will be thrown and caught by the `catch`.

DanielEJaimes replied:
Done

PedroChaparro commented on src/components/UpdatePassword/UpdatePasswordButton.tsx:
```diff
@@ -0,0 +1,136 @@
+              </button>
+              <button
+                type="submit"
+                className="mb-1 mr-1 rounded bg-blue-600 px-6 py-3 text-sm font-bold uppercase text-white shadow outline-none transition-all duration-150 ease-linear hover:shadow-lg focus:outline-none active:bg-emerald-600"
```
This button is different from any other button in the UI. You (both) should have created a custom `<Button />` component to prevent this instead of copying the button styles everywhere you needed a button.

DanielEJaimes replied:
You are right
I created a button for this `ButtonSubmit`

PedroChaparro committed:
chore(deps): Bump dependencies

DanielEJaimes committed:
fix: style

DanielEJaimes committed:
Merge branch 'feat/updatepassword' of https://github.com/hawks-atlanta/frontend-react into feat/updatepassword

DanielEJaimes committed:
fix: cypress test

PedroChaparro commented on src/components/UpdatePassword/UpdatePasswordButton.tsx:
```diff
@@ -0,0 +1,127 @@
+                }
+              })}
+            />
+            <br />
```
Don't use `<br />` to add spacing between elements.

DanielEJaimes committed:
style: Dialog title color

PedroChaparro commented on src/components/UpdatePassword/UpdatePasswordButton.tsx:
```diff
@@ -0,0 +1,127 @@
+      } else {
+        toast.error("Failed to update password");
+      }
+      formRef.current?.reset();
```
It seems that this is not working as expected: 


https://github.com/hawks-atlanta/frontend-react/assets/62714297/d49bf9f4-9778-4e4e-9036-aae1a832d8a8

PedroChaparro commented on src/components/UpdatePassword/UpdatePasswordButton.tsx:
```diff
@@ -0,0 +1,127 @@
+              }`}
+            />
+            <br />
+            {errors.newPassword && (
```
The width of the modal is changing when showing validation errors: 

| ![image](https://github.com/hawks-atlanta/frontend-react/assets/62714297/b8ba2107-a25a-4fa6-aeee-5d08195553fe) | ![image](https://github.com/hawks-atlanta/frontend-react/assets/62714297/bd5fa65d-0b86-4c85-a1ad-a24c68869451) |
|--------|--------|

DanielEJaimes committed:
style: form size and input reset

PedroChaparro reviewed (approved):
Seems to be working. Squash and merge.
